textpipe/doc.py

```diff
@@ -555,3 +558,51 @@ def generate_gensim_document_embedding(self, model_file=None, lowercase=True):
 
         doc_vector = numpy.sum(weighted_word_vectors, axis=0)
         return list(doc_vector)
+
+
+    @functools.lru_cache()
+    def generate_textrank_summary(self, ratio=0.2):
+        """
+        returns a textrank summary of the document (extractive summary) generated with gensim
+        """
+        extracted_sentences = []
+        if self.nsents > 1:
```

Comment:
why is the summary empty if there's only 1 sentence?

Reply:
in principle the summary reduces the number of sentences to the determined ratio. So if the ratio is 0.2 of a text consisting of 1 sentence, the resulting summary would be empty (no summary possible), because the text cannot be shortened (with the textrank algorithm).

Reply:
Shouldn't it then simply return the one sentence?
And how about the situation where we have 2 sentences? Or more generic: what you're saying is that the threshold (1 here) really should depend on the ratio. I guess I agree. If the ration where 1.0, for instance, it would be confusing to return an empty summary.

Reply:
In the case of 2 sentences and a too high ratio, textrank will also return an empty summary if I am correct (I will test that), the reason for the explicit case of 1 sentence is that textrank already does not accept that as input. 

To me the interpretation is more or less, I cannot provide a summary with that compression ratio, so I return no summary at all. You are saying, if that ratio is not possible, then just return the uncompressed version. The downside of that, is that it is not explicit at all that there was no actual compression, which I dislike (you would have to compare lengths of summary and original text to find out that it was not summarized).

I do agree that a ratio of 1.0 would be strange to return no sentences - on the other hand, it would be weird to input a ratio of 1.0 to a summarization algorithm ;) but the option is there, so I will change the code, so that in the case of 1 sentence and a ratio of 1.0 the original sentence is returned, ok?

Reply:
Hmmm, something seems fundamentally wrong then (I mean, what about a ratio of 0.999)

How about we simply pass the input directly to textrank and return whatever it returns. We catch errors that occur and return [] for the invalid input case?